### ISSUE TITLE
@each can only be one level deep

### ISSUE THREAD

**Issue by cah-briangantzler** [miguelcobain/ember-yeti-table#109]
I am getting a deprecation that the @each can only be one level deep and have traced it yeti table.

I have a model that is my inventory. The inventory has a property on it for the product. Product is a master table and several instances of inventory point to the same product. Product contains things like description, manufacturer, color, etc. In the yeti table it is desirable to print some of these attributes. Unfortunately that makes the property `product.description`. Due to line https://github.com/miguelcobain/ember-yeti-table/blob/master/addon/components/yeti-table/component.js#L365 this attempts to add an @each for every property. I would rather not burden the inventory object with a computed `productDescription` and every other property on product to abstract away the relation.  

Since this data is not really going to change, it would be nice if we could add an attribute that would say, dont include this in the @each. @invalidateProperty. if not present use the column prop name. If present use that. That way I could include just `product` to invalidate on the relation changing. And maybe allow false to say dont include this column in the @each.  

A filter on the columns a few lines above could fix that. May also want to include a unique on columns. Someone I guess could have the same prop on two different columns.

I can put together a PR, but wanted to checkin first on what might be a good attribute name for this, or perhaps you had a different approach.

**Comment by miguelcobain** [comment]
@cah-briangantzler What if you use a 
```hbs
<header.column @prop="product">
  Description
</header.column>
```

and then, in the body, you render the description like:
```hbs
<row.cell>
  {{item.product.description}}
</row.cell>
```

The column's `@prop` value is the thing that will be added to the `@each` key.

You can't use yeti table's automatic table unrolling for this, but I guess that the full form is there for cases like this when some customization is needed. Did I miss something?

**Comment by cah-briangantzler** [comment]
Filtering and sorting would not work correctly would it?. I also couldn’t do this for all the props as you would get a duplicate @each and no prop would definitely not work for filtering and sorting.

**Comment by miguelcobain** [comment]
You are correct. Filtering and sorting wouldn't work correctly. In the case of filtering, for example, we use the `@prop` key to get the value used for the filtering operation.

Right now we're directly setting all the columns `prop` values into the `@each` key. What do you think if we:
- be a little smarter and truncate all the `@prop` values up until the first `.`, if it exists. This way we always guarantee that we're using valid keys
- mention in the docs that since ember does not support nested each keys, the table won't update if you change that nested property

**Comment by cah-briangantzler** [comment]
I had thought about truncating to first period, but are we making a decision for them? And if the value really isn’t going to change, would be nice not to create the @each for that column, that’s just putting an observer on every record in the array needlessly. Although truncating to first period would only be one property added to the @each not every property.

Would also have to weed out duplicates by adding a unique after the map.

Im actually fine with truncating but the docs would have to be pretty clear when they find they are changing it, why it wasn’t working. And then that would have to create a computed to abstract away the relation to actually get it to work.

I feel deprecation would happen a lot with any ember data model that has a belongsTo. Funny how I hadn’t noticed it before.

**Comment by cah-briangantzler** [comment]
Hmmm, I should reply in the thread instead of via email

**Comment by cah-briangantzler** [comment]
Once we arrive at a solution, if you want me to do the PR, just let me know

**Comment by cah-briangantzler** [comment]
Thinking about it more I think that the ability to opt out of having the property in the each is a good thing. No sense having ember setting up the observers on properties that you dont need. Would seem a performance improvement. 

Going to vote again for `@invalidateProperty`. When setting up the `@each` if not present use value from `@prop`, if present and false, do not include in `@each`, else use value in `@each`.

For this to matter the table has to be on the screen while the records properties are altered, some kind of editing. However many times the table is used to display only, and the editing could be on another route. In that case, no property needs to be in the `@each`. Might be nice to support `@invalidateProperty` = false at the table level as well.

While we cant wait, assuming that `@tracked` works for properties more than one deep.

**Comment by miguelcobain** [comment]
Fixed on `v1.0.1`. Thanks!
